upstream-community-operators/infinispan/infinispan-operator.v0.1.0.clusterserviceversion.yaml

```diff
@@ -0,0 +1,181 @@
+  version: 0.1.0
+  maturity: stable
+  replaces: infinispan-operator.v0.0.0
+  maintainers:
+    - name: Infinispan Community
+      email: infinispan-dev@lists.jboss.org
+  provider:
+    name: Red Hat
+  labels:
+    operated-by: infinispan-operator
+  selector:
+    matchLabels:
+      operated-by: infinispan-operator
+  links:
+    - name: Blog
+      url: https://blog.infinispan.org/
+    - name: Documentation
+      url: TBD
+    - name: etcd Operator Source Code
+      url: https://github.com/infinispan/infinispan-operator
+
+  icon:
+    - base64data: TBD
+      mediatype: image/png
+  installModes:
+    - type: OwnNamespace
+      supported: true
+    - type: SingleNamespace
+      supported: true
+    - type: MultiNamespace
+      supported: true
+    - type: AllNamespaces
+      supported: true
+  install:
+    strategy: deployment
+    spec:
+      permissions:
+        - serviceAccountName: infinispan-operator
+          rules:
+          - apiGroups:
+              - ""
+            resources:
+              - pods
+              - services
+              - endpoints
+              - persistentvolumeclaims
+              - events
+              - configmaps
+              - secrets
+            verbs:
+              - '*'
+          - apiGroups:
+              - ""
+            resources:
+              - namespaces
+            verbs:
+              - get
+          - apiGroups:
+              - apps
+            resources:
+              - deployments
+              - daemonsets
+              - replicasets
+              - statefulsets
+            verbs:
+              - '*'
+          - apiGroups:
+              - monitoring.coreos.com
+            resources:
+              - servicemonitors
+            verbs:
+              - get
+              - create
+          - apiGroups:
+              - infinispan.org
+            resources:
+              - '*'
+            verbs:
+              - '*'
+      deployments:
+        - name: infinispan-operator
+          spec:
+            replicas: 1
+            selector:
+              matchLabels:
+                name: infinispan-operator-alm-owned
+            template:
+              metadata:
+                name: infinispan-operator-alm-owned
+                labels:
+                  name: infinispan-operator-alm-owned
+              spec:
+                serviceAccountName: infinispan-operator
+                containers:
+                  - name: infinispan-operator
+                    command:
+                      - infinispan-operator
+                    image: jboss/infinispan-operator:0.1.0
+                    ports:
+                      - containerPort: 60000
+                        name: metrics
+                    env:
+                      - name: WATCH_NAMESPACE
+                        valueFrom:
+                          fieldRef:
+                            fieldPath: metadata.namespace
```

Comment:
If this operator supports the downward API for watching cluster resources, then this should be set to `metadata.annotations['olm.targetNamespaces']`

Reply:
We don't yet support watching the cluster resources, what should we put here instead?

Reply:
@SamiSousa I copied `metadata.namespace` from the etc operator. What's the difference between the two values? What is this really for?

Reply:
If this operator doesn't support watching resources clusterwide, please set the `AllNamespaces` installMode to `false`.
This field should be set to `metadata.annotations['olm.targetNamespaces']`. The etcd operator is an antipattern.
Operators should be using OLM's annotation to know which namespace(s) to watch. If the operator is installed with an OperatorGroup that watches one namespace, this annotation will be set to the name of that one namespace. If the OperatorGroup watches multiple namespaces, this annotation will be set to a comma separated list of namespaces that are watched. If the OperatorGroup watches all namespaces, this string is set to the empty string, and the operator must use the downward API to watch CRs clusterwide.
More info can be found here: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/operatorgroups.md